crates/amaru/src/bin/amaru/cmd/import_ledger_state.rs

```diff
@@ -148,12 +149,110 @@ async fn import_all(
     Ok(())
 }
 
+/// Various parts of a full ledger state
+enum StateFragment {
+    EpochNo(Epoch),
+    CurrentBlockIssuers(BTreeMap<PoolId, u64>),
+    Treasury { treasury: i64, reserve: i64 },
+}
+
+/// Simple wrapper over an iterator to enumerate `StateFragment` for
+/// a given epoch
+trait EpochState<'a, E>: Iterator<Item = Result<StateFragment, E>> {
+    /// FIXME: temporary kludge while refactoring code
+    fn decoder(&self) -> minicbor::Decoder<'a>;
```

Comment:
Use `cbor` ?

Reply:
I did not change anything here, just used what was there already.